lua_examples/lfs/_init.lua

```diff
@@ -24,36 +24,40 @@
                  print(table.concat(LFS._list,'\n'))
                gives you a single column listing of all modules in the LFS.
 
+   For Lua 5.3 LFS table is populated by the LFS implementation in C so this part
+   of the code is skipped.
 ---------------------------------------------------------------------------------]]
 
 local index = node.flashindex
-
-local lfs_t = {
-  __index = function(_, name)
-      local fn_ut, ba, ma, size, modules = index(name)
-      if not ba then
-        return fn_ut
-      elseif name == '_time' then
-        return fn_ut
-      elseif name == '_config' then
-        local fs_ma, fs_size = file.fscfg()
-        return {lfs_base = ba, lfs_mapped = ma, lfs_size = size,
-                fs_mapped = fs_ma, fs_size = fs_size}
-      elseif name == '_list' then
-        return modules
-      else
-        return nil
-      end
-    end,
-
-  __newindex = function(_, name, value) -- luacheck: no unused
-      error("LFS is readonly. Invalid write to LFS." .. name, 2)
-    end,
-
+local G=_ENV or getfenv()
+if _VERSION == 'Lua 5.1' then
```

Comment:
We need to set `_G.LFS` to `node.LFS` on the `else` branch,  otherwise `LFS` will not resolve for the require code @ 74.  Though IMO, it would be better to hoist local `lfs_t` which is then set by the `if _VERSION` code as per 5.1 or 5.3 (the 5.1 path includes the `setmetatable()`) and move the `G.LFS = lfs_t` to the common path.  Then L74 can refer to the close upval `lfs_t`.

Reply:
This was before `node.LFS` was implemented.
I like the idea to use upvalue in `package.loaders[3]` handler.